ruby/src/api.cc

```diff
@@ -0,0 +1,509 @@
+#include <leatherman/ruby/api.hpp>
+#include <leatherman/util/environment.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem/path.hpp>
+#include <boost/filesystem/operations.hpp>
+#include <sstream>
+
+using namespace std;
+using namespace leatherman::util;
+using namespace leatherman::execution;
+using namespace boost::filesystem;
+
+namespace lth_lib = leatherman::dynamic_library;
```

Comment:
Which compiler complains, clang? I don't think it should be ambiguous, but I'm likely wrong.